lib/std/channel.nim

```diff
@@ -0,0 +1,700 @@
+#
+#
+#           The Nim Compiler
+#        (c) Copyright 2021 Mratsim & Nim Contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
```

Comment:
this needs a detailed doc comment or at least comment describing how this compares to lib/system/channels.nim, whether it'll evenually support --gc:refc, what to do with lib/system/channels.nim etc

Reply:
It won't support refc. It will replace `lib/system/channels.nim` eventually when `orc` is production-ready.